crates/core/plugin_sm/src/plugin.rs

```diff
@@ -219,14 +219,20 @@ pub struct ExternalPluginCommand {
     pub name: SoftwareType,
     pub path: PathBuf,
     pub sudo: Option<PathBuf>,
+    pub config_location: TEdgeConfigLocation,
 }
 
 impl ExternalPluginCommand {
-    pub fn new(name: impl Into<SoftwareType>, path: impl Into<PathBuf>) -> ExternalPluginCommand {
+    pub fn new(
+        name: impl Into<SoftwareType>,
+        path: impl Into<PathBuf>,
+        config_location: TEdgeConfigLocation,
```

Comment:
I would consider to directly pass the value of `config.software.plugin.max_packages` instead of a config location.

- This is not a big deal but currently all plugin instances will have to read the config only to get this value.
- This also introduce a dependency to `tedge_config`.